packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-credit-button-strategy.ts

```diff
@@ -40,17 +49,32 @@ export default class PaypalCommerceCreditButtonStrategy implements CheckoutButto
         const currencyCode = state.cart.getCartOrThrow().currency.code;
         const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
         this._paypalCommerceSdk = await this._paypalScriptLoader.getPayPalSDK(paymentMethod, currencyCode, initializesOnCheckoutPage);
-
-        this._renderButton(containerId, methodId, initializesOnCheckoutPage, style);
+        const { initializationData } = paymentMethod;
+        const { isHostedCheckoutEnabled } = initializationData;
+        this._renderButton(containerId, methodId, isHostedCheckoutEnabled, initializesOnCheckoutPage, style);
         this._renderMessages(messagingContainerId);
     }
 
     deinitialize(): Promise<void> {
         return Promise.resolve();
     }
 
-    private _renderButton(containerId: string, methodId: string, initializesOnCheckoutPage?: boolean, style?: PaypalButtonStyleOptions): void {
+    private _renderButton(containerId: string, methodId: string, isHostedCheckoutEnabled: boolean, initializesOnCheckoutPage?: boolean, style?: PaypalButtonStyleOptions): void {
         const paypalCommerceSdk = this._getPayPalCommerceSdkOrThrow();
+
+        const hostedCallbacks = {
```

Comment:
It might be easier to understand if we update `hostedCallbacks` to `hostedCheckoutCallbacks`

```suggestion
        const hostedCheckoutCallbacks = {
```

Reply:
renamed